app/services/reporting.js

```diff
@@ -211,4 +240,112 @@ export default class ReportingService extends Service {
     }).sort();
     return [[this.intl.t('general.vocabulary')]].concat(titles);
   }
+
+  async buildReportTitle(report) {
+    const subject = report.subject;
+    const key = subjectTranslations[subject];
+    const subjectTranslation = this.intl.t(key);
+    const prepositionalObject = report.prepositionalObject;
+
+    const school = await report.school;
+    const schoolTitle = school ? school.title : this.intl.t('general.allSchools');
+
+    if (prepositionalObject) {
+      let model = dasherize(prepositionalObject);
+      if (model === 'instructor') {
+        model = 'user';
+      }
+      if (model === 'mesh-term') {
+        model = 'mesh-descriptor';
+      }
+      const prepositionalObjectTableRowId = report.get('prepositionalObjectTableRowId');
+
+      let record;
+      try {
+        record = await this.store.findRecord(model, prepositionalObjectTableRowId);
+      } catch (e) {
+        return this.intl.t('general.thisReportIsNoLongerAvailable');
+      }
+
+      let object;
+      if (model === 'user') {
+        object = record.fullName;
+      } else if (model === 'mesh-descriptor') {
+        object = record.name;
+      } else {
+        object = record.title;
+      }
+
+      return this.intl.t('general.reportDisplayTitleWithObject', {
+        subject: subjectTranslation,
+        object,
+        school: schoolTitle,
+      });
+    }
+
+    return this.intl.t('general.reportDisplayTitleWithoutObject', {
+      subject: subjectTranslation,
+      school: schoolTitle,
+    });
+  }
+
+  async buildReportDescription(report) {
```

Comment:
I think this is nearly identical to `buildReportTitle`, can the loading be refactored into a third method and then these two methods just do the translation?

Reply:
:heavy_check_mark: